modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinTcpIo.java

```diff
@@ -319,6 +335,13 @@ private HandshakeResult handshake(ClientListenerProtocolVersion ver) throws IOEx
         if (ver.compareTo(VER_2_13_0) >= 0)
             writer.writeString(connProps.getQueryEngine());
 
+        if (ver.compareTo(VER_2_14_0) >= 0) {
+            writer.writeByte(connProps.getTransactionConcurrency().ordinal());
```

Comment:
Why not reuse JdbcSetTxParametersRequest message (and send it only when TX_AWARE_QUERIES is enabled)  without changing handshake and protocol version?

Reply:
Add of transaction parameters to handshake was requested by @timoninmaxim to reduce network messages.